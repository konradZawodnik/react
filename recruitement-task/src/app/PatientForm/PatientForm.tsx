"use client";
import { useForm } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";
import { z } from "zod";
import { Button } from "@/components/ui/button";
import { Label } from "@/components/ui/label";
import { ChevronRight } from "lucide-react";
import { Form } from "@/components/ui/form";
import VisitData from "./VisitData/VisitData";
import { VisitDataType } from "../types/types";

export default function PatientForm() {

  const onSubmit = (data: VisitDataType) => {
    return data;
  };

  const fieldSchema = z.object({
    age: z.string({
      required_error: "Proszę wybrać wiek",
    }),
    addressCountry: z.string({
      required_error: "Proszę wybrać kraj",
    }),
    addressStreet: z.string({
      required_error: "Proszę wpisać adres",
    }),
    addressNumber: z.string({
      required_error: "Proszę wpisać numer adresu",
    }),
    firstName: z.string({
      required_error: "Proszę podać imię pacjenta",
    }),
    isAnotherVisitAddress: z.boolean().default(false).optional(),
    peselNumber: z
      .string({
        required_error: "Niewłaściwy numer PESEL",
      })
      .default("")
      .optional(),
    passportNumber: z
      .string({
        required_error: "Niewłaściwy numer PESEL",
      })
      .default("")
      .optional(),
    surname: z.string({
      required_error: "Proszę podać nazwisko pacjenta",
    }),
    symptoms: z.string().default("").optional(),
    visitAddressCountry: z.string().default("").optional(),
    visitAddressStreet: z.string().default("").optional(),
    visitAddressNumber: z.string().default("").optional(),
  });

  const FormSchema = z.object({
    additionalInformations: z.string().default("").optional(),
    isTimeSlotChecked: z.boolean().default(false).optional(),
    language: z.string({
      required_error: "Proszę wybrać język",
    }),
    fields: z.array(fieldSchema),
    reportNumber: z.string({
      required_error: "Proszę podać numer zgłoszenia",
    }),
    specialization: z.string({
      required_error: "Proszę wybrać specjalizację",
    }),
    topic: z.string({
      required_error: "Proszę wybrać temat",
    }),
    visitDate: z.date({
      required_error: "Proszę wybrać datę wizyty",
    }),
    visitFrom: z.string().default("").optional(),
    visitTo: z.string().default("").optional(),
    visitType: z.string({
      required_error: "Proszę wybrać rodzaj wizyty",
    }),
  });

  const form = useForm<z.infer<typeof FormSchema>>({
    resolver: zodResolver(FormSchema),
    mode: "onChange",
  });

  //TODO Rest of form

  return (
    <div className="flex flex-col items-start ml-2 mr-4 w-full">
      <Label className="flex items-start font-[400] mb-2 mt-8 pl-4 text-[#6d7178] text-[14px] w-full">
        Wizyty domowe{" "}
        <ChevronRight className="mt-[2px]" height={18} width={18} /> Umawianie
        wizyty
      </Label>
      <Label className="flex items-start font-[300] mb-2 pl-4 text-[#112950] text-4xl w-full">
        Umawianie wizyty
      </Label>
      <Form {...form}>
        <div className="bg-white flex flex-col ml-4 mr-4 p-16 max-w-[715px] w-full">
          <Label className="font-[300] leading-7 mb-4 text-[24px] text-[#112950]">
            Wizyta
          </Label>
          <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-6">
            <VisitData
              control={form?.control}
              isTimeSlotChecked={form.getValues().isTimeSlotChecked}
            />

            <div className="flex flex-col justify-end w-full">
              <Button
                className="bg-[#0068fa] hover:bg-[#0068fa] ml-0 mt-4 text-white w-full"
                type="submit"
              >
                Dalej <ChevronRight />
              </Button>
            </div>
          </form>
        </div>
      </Form>
    </div>
  );
}
