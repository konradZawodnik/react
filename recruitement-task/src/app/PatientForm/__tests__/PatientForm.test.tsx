import { render } from "@testing-library/react";
import PatientForm from "../PatientForm";

global.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));

it("renders patientForm unchanged", () => {
  const { container } = render(<PatientForm />);
  expect(container).toMatchSnapshot();
});
