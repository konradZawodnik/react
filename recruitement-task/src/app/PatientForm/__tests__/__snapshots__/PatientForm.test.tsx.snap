// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders patientForm unchanged 1`] = `
<div>
  <div
    class="flex flex-col items-start ml-2 mr-4 w-full"
  >
    <label
      class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 flex items-start font-[400] mb-2 mt-8 pl-4 text-[#6d7178] text-[14px] w-full"
    >
      Wizyty domowe
       
      <svg
        class="lucide lucide-chevron-right mt-[2px]"
        fill="none"
        height="18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
       Umawianie wizyty
    </label>
    <label
      class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 flex items-start font-[300] mb-2 pl-4 text-[#112950] text-4xl w-full"
    >
      Umawianie wizyty
    </label>
    <div
      class="bg-white flex flex-col ml-4 mr-4 p-16 max-w-[715px] w-full"
    >
      <label
        class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-[300] mb-4 text-[24px] text-[#112950]"
      >
        Wizyta
      </label>
      <form
        class="space-y-6"
      >
        <div
          class="space-y-4 w-full"
        >
          <div
            class="space-y-2"
          >
            <label
              class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-[700] text-[#112950]"
              for=":r0:-form-item"
            >
              Numer zgłoszenia
            </label>
            <input
              aria-describedby=":r0:-form-item-description"
              aria-invalid="false"
              class="flex h-9 w-full rounded-md border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm border-0"
              id=":r0:-form-item"
              name="reportNumber"
              placeholder="Wpisz numer zgłoszenia"
              type="number"
            />
          </div>
        </div>
        <div
          class="space-y-4 w-full"
        >
          <div
            class="space-y-2"
          >
            <label
              class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-[700] text-[#112950]"
              for=":r1:-form-item"
            >
              Rodzaj wizyty
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-:r2:"
              aria-expanded="false"
              class="flex h-9 items-center justify-between whitespace-nowrap rounded-md border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 border-0 w-full"
              data-placeholder=""
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                Wybierz z listy
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
            <select
              aria-hidden="true"
              name="visitType"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            >
              <option
                value=""
              />
              <option
                value="homeVisit"
              >
                Wizyta domowa
              </option>
              <option
                value="onlineVisit"
              >
                Wizyta online
              </option>
              <option
                value="stationaryVisit"
              >
                Wizyta stacjonarna
              </option>
            </select>
          </div>
        </div>
        <div
          class="space-y-4 w-full"
        >
          <div
            class="space-y-2"
          >
            <label
              class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-[700] text-[#112950]"
              for=":r3:-form-item"
            >
              Specjalizacja
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-:r4:"
              aria-expanded="false"
              class="flex h-9 items-center justify-between whitespace-nowrap rounded-md border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 border-0 w-full"
              data-placeholder=""
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                Wybierz z listy
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
            <select
              aria-hidden="true"
              name="specialization"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            >
              <option
                value=""
              />
              <option
                value="general"
              >
                Ogólna
              </option>
              <option
                value="cardiology"
              >
                Kardiolog
              </option>
              <option
                value="neurology"
              >
                Neurolog
              </option>
            </select>
          </div>
        </div>
        <div
          class="flex flex-col space-y-4 w-full"
        >
          <div
            class="space-y-2 flex flex-col"
          >
            <label
              class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-[700] text-[#112950]"
              for=":r5:-form-item"
            >
              Data wizyty
            </label>
            <button
              aria-controls="radix-:r6:"
              aria-describedby=":r5:-form-item-description"
              aria-expanded="false"
              aria-haspopup="dialog"
              aria-invalid="false"
              class="inline-flex items-center gap-2 whitespace-nowrap rounded-md text-sm transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 border-0 w-full justify-start text-left font-normal text-muted-foreground"
              data-state="closed"
              id=":r5:-form-item"
              type="button"
            >
              <svg
                class="lucide lucide-calendar mr-2 h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 2v4"
                />
                <path
                  d="M16 2v4"
                />
                <rect
                  height="18"
                  rx="2"
                  width="18"
                  x="3"
                  y="4"
                />
                <path
                  d="M3 10h18"
                />
              </svg>
              <span>
                Jak najszybciej
              </span>
            </button>
          </div>
        </div>
        <div
          class="flex flex-row w-full"
        >
          <div
            class="space-y-2 flex flex-row items-center"
          >
            <button
              aria-checked="false"
              aria-describedby=":r7:-form-item-description"
              aria-invalid="false"
              class="peer h-4 w-4 shrink-0 rounded-sm border border-primary shadow focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
              data-state="unchecked"
              id=":r7:-form-item"
              role="checkbox"
              type="button"
              value="on"
            />
            <input
              aria-hidden="true"
              style="transform: translateX(-100%); position: absolute; pointer-events: none; opacity: 0; margin: 0px; width: 0px; height: 0px;"
              tabindex="-1"
              type="checkbox"
              value="on"
            />
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 border-0 font-[400] ml-2 mt-0 text-[14px] text-[#112950]"
              for="isTimeSlotChecked"
            >
              Wybierz konkrenty przedział godzinowy
            </label>
          </div>
        </div>
        <div
          class="space-y-4 w-full"
        >
          <div
            class="space-y-2"
          >
            <label
              class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-[700] text-[#112950]"
              for=":r8:-form-item"
            >
              Temat
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-:r9:"
              aria-expanded="false"
              class="flex h-9 items-center justify-between whitespace-nowrap rounded-md border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 border-0 w-full"
              data-placeholder=""
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                Wybierz z listy
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
            <select
              aria-hidden="true"
              name="topic"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            >
              <option
                value=""
              />
              <option
                value="healthCheck"
              >
                Badania
              </option>
              <option
                value="emergency"
              >
                Wypadek
              </option>
              <option
                value="followUp"
              >
                Konsultacja
              </option>
            </select>
          </div>
        </div>
        <div
          class="space-y-4 w-full"
        >
          <div
            class="space-y-2"
          >
            <label
              class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 border-0 font-[700] text-[#112950]"
              for=":ra:-form-item"
            >
              Dodatkowe informacje (opcjonalnie)
            </label>
            <textarea
              aria-describedby=":ra:-form-item-description"
              aria-invalid="false"
              class="flex min-h-[60px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-base shadow-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
              id=":ra:-form-item"
              name="additionalInformations"
              placeholder="Opisz problem"
            />
          </div>
        </div>
        <div
          class="space-y-4 w-full"
        >
          <div
            class="space-y-2"
          >
            <label
              class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-[700] text-[#112950]"
              for=":rb:-form-item"
            >
              Język wizyty
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-:rc:"
              aria-expanded="false"
              class="flex h-9 items-center justify-between whitespace-nowrap rounded-md border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 border-0 w-full"
              data-placeholder=""
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                Wybierz z listy
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
            <select
              aria-hidden="true"
              name="language"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            >
              <option
                value=""
              />
              <option
                value="english"
              >
                Angielski
              </option>
              <option
                value="polish"
              >
                Polski
              </option>
              <option
                value="spanish"
              >
                Hiszpański
              </option>
            </select>
          </div>
        </div>
        <div
          class="flex flex-col justify-end w-full"
        >
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 shadow h-9 px-4 py-2 bg-[#0068fa] hover:bg-[#0068fa] ml-0 mt-4 text-white w-full"
            type="submit"
          >
            Dalej 
            <svg
              class="lucide lucide-chevron-right"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
